src/main/java/app/linguistai/bmvp/service/gamification/UserStreakService.java

```diff
@@ -42,15 +46,17 @@ private RUserStreak checkUserStreakForUpdate(RUserStreak rUserStreak) throws Exc
             Date streakTime = DateUtils.convertSqlDateToUtilDate(rUserStreak.getLastLogin());
 
             if (streakTime == null) {
-                throw new Exception("StreakTime is null");
+                throw new NotFoundException("StreakTime", true);
```

Comment:
Moreover, this should not be a `NotFoundException`, as for this case the **UserStreak is there**. The problem as that a `NotNull` field is actually `Null`.